### PR TITLE
Support variadic constructor params in TypedPropertyFromStrictConstructorRector

### PR DESCRIPTION
**Pull request by ruudk**
# Failing Test for TypedPropertyFromStrictConstructorRector

Based on https://getrector.org/demo/331db50e-de84-4ffc-99cb-ae299c427411

### CODE REVIEW

**Comment by ruudk** [comment]
@TomasVotruba I think I fixed the issue 🎉 😎  WDYT?

File: rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php [hunk 1]
```diff
@@ -61,7 +62,13 @@ public function inferProperty(Property $property): Type
 
         // A. infer from type declaration of parameter
         if ($param->type !== null) {
-            return $this->resolveFromParamType($param, $classMethod, $propertyName);
+            $type = $this->resolveFromParamType($param, $classMethod, $propertyName);
+
+            if ($param->variadic) {
+                return new ArrayType(new ConstantIntegerType(0), $type);
```
Comment (ruudk) [inline rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php:68]: Or should it be ?

```suggestion
                return new ArrayType(new IntegerType(), $type);
```

Comment (TomasVotruba) [reply rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php:68]: The `resolveFromParamType()` method should be fixed instead, as it's output is incorrect

Comment (ruudk) [reply rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php]: Like this https://github.com/rectorphp/rector/pull/5359/commits/e8bb282c30dbf3a58d35bb09ef60a5add5ded3a8 ?

Comment (TomasVotruba) [reply rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php]: Super :)

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Comment by ruudk** [comment]
@TomasVotruba 🔥  Wanted to say thank you for this amazing flow on handling a bug. Run code in playground, click "Create test", get a PR that fails, try to commit to fix it, its green, merged, ⚡ 👏

**Comment by TomasVotruba** [comment]
Cool! I'm glad you like it :+1: 

I'm very focused on DX, as it's the main productivity killer in coding. 
If you have any tips, let me know ;)

<br>

Btw, are you still on Phalcon team? We're currently migrating one project for last year and it's a hell to run Rector on C code :D

**Comment by ruudk** [comment]
> Btw, are you still on Phalcon team? We're currently migrating one project for last year and it's a hell to run Rector on C code :D

I think you have confused me with somebody else ;)

**Comment by TomasVotruba** [comment]
Ah, you're right. I thought you're the only Ruud I know. Just checked the legacy history in my mail. My mistake :)

What fw do you work with?

**Comment by ruudk** [comment]
Symfony :)

**Comment by TomasVotruba** [comment]
Good choice :)

**Pull request merged**